src/modules/BuilderPlace/actions/discord.ts

```diff
@@ -0,0 +1,148 @@
+  markdownString = markdownString.replace(/`{3}.*`{3}/g, '');
+
+  // Remove inline code (`)
+  markdownString = markdownString.replace(/`{1}.*`{1}/g, '');
+
+  // Remove links ([...](...))
+  markdownString = markdownString.replace(/\[([^\]]+)\]\([^\)]+\)/g, '$1');
+
+  // Remove images (![...](...))
+  markdownString = markdownString.replace(/!\[([^\]]+)\]\([^\)]+\)/g, '$1');
+
+  return markdownString;
+}
+
+export const sendWelcomeMessage = async (builderPlaceId: string, newWebhookUrl: string) => {
+  const webhookDetails = await getDiscordWebhookDetailsByBuilderPlaceId(builderPlaceId)
+  console.log(webhookDetails, newWebhookUrl, "ah")
+
+  if (webhookDetails?.discordWebhook === newWebhookUrl) return;
+  console.log("sendit", newWebhookUrl, "ah")
+
+  const embed = {
+    title: 'Congratulations!',
+    description: 'BuilderPlace notification is set up. I will notify you here for each new service created!',
+    color: 0x00ff00,
+    author: {
+      name: 'BuilderPlace',
+      icon_url: 'https://builder.place/logo.png',
+    },
+  };
+
+  const payload = {
+    content: ' ', // Empty content to trigger the embed
+    embeds: [embed],
+  };
+
+  fetch(newWebhookUrl, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(payload),
+  })
+    .then((response) => {
+      if (response.ok) {
+        console.log('Message sent successfully!');
+      } else {
+        console.error('Error sending message:', response.statusText);
+      }
+    })
+    .catch((error) => {
+      console.error('Error:', error.message);
+    });
+}
+
+
+export const sendNewServiceNotification = async (builderPlaceId: string, serviceId: string, cid: string) => {
+  const service: { title: string, about: string, keywords: string, rateToken: string, rateAmount: string } = await readFromIPFS(cid);
+  const webhookDetails = await getDiscordWebhookDetailsByBuilderPlaceId(builderPlaceId);
+
+  if (!webhookDetails?.discordWebhook) return;
+
+  const builderPlaceUrl = webhookDetails.customDomain ? webhookDetails.customDomain : webhookDetails.subdomain;
+
+  let allowedTokens: IToken[] = []
+  const allowedTokensResponse = await getAllowedTokenList(process.env.NEXT_PUBLIC_DEFAULT_CHAIN_ID as unknown as number);
+  if (allowedTokensResponse?.data?.data?.tokens) {
+    allowedTokens = allowedTokensResponse?.data?.data?.tokens
+  }
+  else {
+    return;
+  }
+
+  const token: IToken | undefined = allowedTokens.find(token => token.address === service.rateToken);
+  if (!token) return;
+
+  const embed = {
+    title: service.title,
+    description: "New gig, hot and ready for proposals!",
+    color: 2326507,
+    fields: [
+      {
+        name: "Description",
+        value: removeMarkdown(truncate(service.about, 420)),
+        inline: false
+      },
+      {
+        name: "Budget",
+        value: `${(Number(service.rateAmount) / Math.pow(10, token.decimals)).toString()} ${token.symbol}`,
```

Comment:
we have helper normally for that

Reply:
i could not find one, i tried searching the code base for  "Math.pow" and nothing come up.